benchexec/tools/metaval.py

```diff
@@ -108,14 +111,22 @@ def cmdline(self, executable, options, tasks, propertyfile=None, rlimits={}):
                 ).Tool()
 
         if verifierName in self.wrappedTools:
-            with self._in_tool_directory(verifierName) as oldcwd:
-                wrappedOptions = self.wrappedTools[verifierName].cmdline(
-                    self.wrappedTools[verifierName].executable(),
-                    options,
-                    [os.path.relpath(os.path.join(oldcwd, "output/ARG.c"))],
-                    os.path.relpath(os.path.join(oldcwd, propertyfile)),
-                    rlimits,
+            tool = self.wrappedTools[verifierName]
```

Comment:
This variable is used only once, wouldn't it make sense to also use it in the remaining places where the dict is accessed in the following code?

Reply:
Sure, we can do this. This is actually more feedback than I want. It doesn't have to be super perfect and without spaces for improvement. It is my responsibility anyway to maintain this tool-info module, so please let it up to me to decide when it makes sense improving the code structure.

Reply:
Sorry, but I will not stop giving helpful feedback when I notice something.

> It is my responsibility anyway to maintain this tool-info module, so please let it up to me to decide when it makes sense improving the code structure.

This is not true. Tool-info modules are not the sole responsibility of the original submitter. BenchExec developers always have and will continue to also work on these modules, for example to improve code quality (as has happened for `metaval.py` in the past) or even to add features (example: #646). There might also be other developers/maintainers of a tool (and thus of the tool-info module) in the future. So we do require a certain level of code quality.

Also, you are using strawman arguments here, please do not do this:

> It doesn't have to be super perfect and without spaces for improvement.

Nobody has required or expected the code to be perfect.

> not misuse it for forcing me to do refactorings

(cited here from other comment)

Nobody is forcing you to do this. I carefully worded all my suggestions for code improvements as suggestions.

Reply:
> Nobody is forcing you to do this

Well it is my job, and my boss wants this to be done. Otherwise I would totally ditch it.  
I am just under a lot of stress currently because of the SV-COMP related activities. I would really welcome if you respected my wishes here and concentrated on the most important things. I also need to fix the tool. This tool-info business probably costs me more time than working on the actual tool.

But OK, it is your decision and I cannot force you not to give feedback. So I will just try my best to address everything to get it over with. I am just generally frustrated with the amount of time that is wasted with this totally trivial manner.

For next year (if there is a next year with MetaVal) I will move all the MetaVal-specific code into MetaVal itself and do the wrapped verifier handling there, so the tool-info module will become minimal and I won't have to deal with PRs like this one anymore. You need to be aware that the experience with working on tool-info modules is far from - and never has been - pleasant for me.

> Also, you are using strawman arguments here,

Of course I am using exaggerations like any normal being,you have to read between the lines and don't take phrases like that literally! I wanted to signal you that my priorities currently are different to yours. I would rather get the PR merged in a state where the tool-info module is working/functional, then work on the tool and later come back to work on an issue where the quality improvements that could be done are listed.

It is not like I am gone after this PR is merged, as it might be with random one-time contributors. But I already rage-closed this PR (and reopened it after I calmed down) because it is really pulling me down.

Reply:
The original change that was requested in this thread has been implemented, so I will resolve the thread.